### PR TITLE
calculate angular search from particle diameter

### LINKED ISSUE #164: provide a feature to go from the particle diameter to the 'correct' sampling and the pixel size (or low pass filter)
>Very excited to see this one!
We could also provide a feature to let users input the particle diameter, and calculatie the 'correct' sampling from that and  the pixel size (or low pass filter)

_Originally posted by @McHaillet in https://github.com/SBC-Utrecht/pytom-match-pick/issues/163#issuecomment-2093078631_

### PR DESCRIPTION
**Pull request by McHaillet**
Closes #164 

Addition of particle diameter to calculate angular search.

### CODE REVIEW

File: src/pytom_tm/entry_points.py [hunk 1]
```diff
@@ -555,12 +555,25 @@ def match_template(argv=None):
...
     )
     rotation_group = parser.add_argument_group('Angular search')
+    rotation_group.add_argument(
+        "--particle-diameter",
+        type=float,
+        required=False,
+        action=LargerThanZero,
+        help="Provide a particle diameter (in Angstrom) to automatically determine the "
+             "angular sampling using the Crowther criterion. For the max resolution "
+             "the pixel size is used unless a low-pass filter is specified, in which "
```
Comment (sroet) [inline src/pytom_tm/entry_points.py:565]: it is pixelsize * 2, right?
```suggestion
             "the 2 * pixel size is used unless a low-pass filter is specified, in which "
```

File: src/pytom_tm/tmjob.py [hunk 2]
```diff
@@ -332,6 +336,13 @@ def __init__(
         self.start_slice = 0
         self.steps_slice = 1
         self.rotational_symmetry = rotational_symmetry
+        self.particle_diameter = particle_diameter
+        # calculate increment from particle diameter
+        if angle_increment is None and particle_diameter is not None:
```
Comment (sroet) [inline src/pytom_tm/tmjob.py:341]: should we update the default angle increment to `None` instead of `7.0` and set it to `7.0` during the init?

**Review by sroet — APPROVED** [review]
one help question and a suggestion for updated defaults, feel free to ignore. LGTM otherwise

**Comment by McHaillet** [comment]
I'm merging!

**Pull request merged**